### ISSUE TITLE
Directories with spaces in .formatting-directory being ignored

### ISSUE THREAD

**Issue by notjosh** [square/spacecommander#23]
I can't tell if this is Bash witchcraft, or intended behaviour, or what.

Looking at https://github.com/square/spacecommander/blob/master/lib/common-lib.sh#L12, we see that `directories_to_check()` replaces spaces (`' '`) with escaped spaces (`'\ '`) via a `sed` command, which seems like a good idea Because Bash™.

BUT tracing the output, it seems that Bash will automatically quote the paths automatically (?!), and as such it will be looking for a string with a literal '`\ `' in its path. Obviously it won't find that!

### Detail

I've cut the `.formatting-directory` down to the following:

```
Clue/Tests
Clue/Clue WatchKit Extension
```

Then running the `format-objc-files-in-repo.sh`, I get the following output:

```
$ Submodules/spacecommander/format-objc-files-in-repo.sh
+ IFS='
'
++ date +%s
+ start_date=1446651719
+ export CDPATH=
+ CDPATH=
+++ dirname Submodules/spacecommander/format-objc-files-in-repo.sh
++ cd Submodules/spacecommander
++ pwd
+ DIR=/Users/joshua/dev/work/Clue/wheel-iOS/Submodules/spacecommander
+ source /Users/joshua/dev/work/Clue/wheel-iOS/Submodules/spacecommander/lib/common-lib.sh
++ IFS='
'
++ all_valid_objc_files_in_repo
++ directories_to_check
++ locations_to_diff=
++ '[' -e .formatting-directory ']'
+++ cat .formatting-directory
+++ sed 's/ /\\ /g'
++ locations_to_diff='Clue/Tests
Clue/Clue\ WatchKit\ Extension'
+++ git ls-tree --name-only --full-tree -r HEAD -- Clue/Tests 'Clue/Clue\ WatchKit\ Extension'
+++ grep -e '\.m$' -e '\.mm$' -e '\.h$' -e '\.hh$'
++ files='Clue/Tests/BWAPICredentialServiceTest.m
Clue/Tests/BWAPIRequestOperationManagerTest.m
Clue/Tests/BWAPIServiceTest.m
...
Clue/Tests/TagParsingTest.m'
++ grep -v Pods/
++ grep -v Carthage/
+ objc_files='Clue/Tests/BWAPICredentialServiceTest.m
Clue/Tests/BWAPIRequestOperationManagerTest.m
Clue/Tests/BWAPIServiceTest.mtionManagerTest.m
Clue/Tests/BWAPIServiceTest.m
...
Clue/Tests/TagParsingTest.m'
+ '[' -z 'Clue/Tests/BWAPICredentialServiceTest.m
Clue/Tests/BWAPIRequestOperationManagerTest.m
Clue/Tests/BWAPIServiceTest.m
...
Clue/Tests/TagParsingTest.m' ']'
...
+ '[' -z '' ']'
++ date +%s
+ end_date=1446651719
+ time_diff=0
+ echo '0 minutes and 0 seconds to format objc.'
0 minutes and 0 seconds to format objc.
+ exit 0
```

Specifically the lines:

```
++ locations_to_diff='Clue/Tests
Clue/Clue\ WatchKit\ Extension'
+++ git ls-tree --name-only --full-tree -r HEAD -- Clue/Tests 'Clue/Clue\ WatchKit\ Extension'
```

See the escaping in the quoted string? That's not the best of situations :crocodile: 

The fix I'm using, but I can't tell if it's a totally stupid idea (or else I'd send a PR!) is to remove the sed, and then I get the output as expected:

```
...
++ '[' -e .formatting-directory ']'
+++ cat .formatting-directory
++ locations_to_diff='Clue/Tests
Clue/Clue WatchKit Extension'
+++ git ls-tree --name-only --full-tree -r HEAD -- Clue/Tests 'Clue/Clue WatchKit Extension'
...
++ files='Clue/Clue WatchKit Extension/BWWKEApplication.h
Clue/Clue WatchKit Extension/BWWKEApplication.m
Clue/Tests/BWAPICredentialServiceTest.m
Clue/Tests/BWAPIRequestOperationManagerTest.m
...
```

### So basically..

Can we drop that `sed` command and trust Bash to do the Right Thing™? Is that a flag option, a version thing, an OS thing?

**Comment by alanf** [comment]
Thanks for the report! Will look into this and see what's up.

**Comment by alanf** [comment]
I agree that this escaping may have been over-cautionary. After checking it out, I think we should go ahead with the PR as you described, but wanted to check in with @ksuther first to see if he remembered how necessary that `sed` was. Thanks again for the detailed writeup!

**Comment by ksuther** [comment]
This sounds correct to me, I'm seeing the same issue as well. I must have had some situation where I thought this was necessary. Maybe I started out with that sed and then made other changes elsewhere that rendered it unnecessary? Looks wrong now though, oops!

**Comment by alanf** [comment]
Thanks @ksuther. @notjosh, feel free to create a PR if you like, otherwise I can get to that later on.

**Comment by notjosh** [comment]
Great! Thanks for confirming that. PR opened at #24 :tada:
